crates/ruff_linter/src/rules/ruff/rules/sort_dunder_all.rs

```diff
@@ -0,0 +1,499 @@
+///     __all__ += ["z", "y"]
+/// ```
+///
+/// Use instead:
+/// ```python
+/// import sys
+///
+/// __all__ = [
+///     "a",
+///     "b",
+///     "c",
+/// ]
+///
+/// if sys.platform == "win32":
+///     __all__ += ["y", "z"]
+/// ```
+///
+/// ## Fix safety
+/// This rule's fix should be safe for single-line `__all__` definitions
+/// and for multiline `__all__` definitions without comments.
+/// For multiline `__all__` definitions that include comments,
+/// the fix is marked as unsafe, as it can be hard to tell where the comments
+/// should be moved to when sorting the contents of `__all__`.
+#[violation]
+pub struct UnsortedDunderAll;
+
+impl Violation for UnsortedDunderAll {
+    const FIX_AVAILABILITY: FixAvailability = FixAvailability::Sometimes;
+
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("`__all__` is not alphabetically sorted")
+    }
+
+    fn fix_title(&self) -> Option<String> {
+        Some("Sort `__all__` alphabetically".to_string())
+    }
+}
+
+pub(crate) fn sort_dunder_all_assign(
+    checker: &mut Checker,
+    node: &ast::StmtAssign,
+    parent: &ast::Stmt,
+) {
+    let ast::StmtAssign { value, targets, .. } = node;
+    let [ast::Expr::Name(ast::ExprName { id, .. })] = targets.as_slice() else {
+        return;
+    };
+    sort_dunder_all(checker, id, value, parent);
+}
+
+pub(crate) fn sort_dunder_all_augassign(
+    checker: &mut Checker,
+    node: &ast::StmtAugAssign,
+    parent: &ast::Stmt,
+) {
+    let ast::StmtAugAssign {
+        value,
+        target,
+        op: ast::Operator::Add,
+        ..
+    } = node
+    else {
+        return;
+    };
+    let ast::Expr::Name(ast::ExprName { id, .. }) = target.as_ref() else {
+        return;
+    };
+    sort_dunder_all(checker, id, value, parent);
+}
+
+pub(crate) fn sort_dunder_all_annassign(
+    checker: &mut Checker,
+    node: &ast::StmtAnnAssign,
+    parent: &ast::Stmt,
+) {
+    let ast::StmtAnnAssign {
+        target,
+        value: Some(val),
+        ..
+    } = node
+    else {
+        return;
+    };
+    let ast::Expr::Name(ast::ExprName { id, .. }) = target.as_ref() else {
+        return;
+    };
+    sort_dunder_all(checker, id, val, parent);
+}
+
+fn sort_dunder_all(checker: &mut Checker, target: &str, node: &ast::Expr, parent: &ast::Stmt) {
+    if target != "__all__" {
+        return;
+    }
+
+    // We're only interested in `__all__` in the global scope
+    if !checker.semantic().current_scope().kind.is_module() {
+        return;
+    }
+
+    let locator = checker.locator();
+
+    let Some(dunder_all_val) = DunderAllValue::from_expr(node, locator) else {
+        return;
+    };
+
+    let new_dunder_all =
+        match dunder_all_val.construct_sorted_all(locator, parent, checker.stylist()) {
+            SortedDunderAll::AlreadySorted => return,
+            SortedDunderAll::Sorted(value) => value,
+        };
+
+    let dunder_all_range = dunder_all_val.range();
```

Comment:
Personal style, I'd probably just inline this wherever it's used, I don't know that the binding adds much but it does add more things named `dunder_all_*` to worry about.